src/client/containers/BoardPageComponents/BoardAddCandidate/AddCandidate.js

```diff
@@ -0,0 +1,43 @@
+// import React, { useState } from 'react';
+import postData from './PostData';
+
+export default async function AddCandidate(
+  newCandidate,
+  userId,
+  boardId,
+  setCandidates,
+  candidates,
+  setAddCandidateError,
+  setAddCandidateSuccess,
+) {
+  try {
+    if (!newCandidate.name) {
+      setAddCandidateError('Please put a candidate name');
+      return;
+    }
+
+    if (!isNaN(newCandidate.name)) {
+      setAddCandidateError('Candidate name can not be number');
+      return;
+    }
+
+    postData(`/api/users/${userId}/boards/${boardId}/candidates`, {
+      name: newCandidate.name,
+    });
+    setAddCandidateSuccess(true);
+    setAddCandidateError(null);
+
+    setCandidates((prev) => {
```

Comment:
Remove the setCandidates. There is no candidates state to update here. Once the the upload has completed. The board will reload and fetch the new list of candidates

Reply:
This component uploads a single candidate. It know nothing of the other candidates.
Please create a CandidateList container component for the BoardPageComponents. This component will contain a list of the candidates and send PUT requests when the ordering changes

Reply:
...And so the BoardComponent knows nothing of candidatelists and adding candidates. It only knows they exists but thats it. The components reside in the BoardComponent but it has no reference to any of the details in the two subcomponents.

Reply:
**"The high level components know nothing of the details of the lower level components"**

Reply:
Thanks Niels, very helpful explanations

Reply:
@senner007 Hi Niels, may I ask how does the CandidateList container get the candidates data. Is it coming from props or context provider or useCandidates hook inside this container?

Reply:
You could make the call in a hook, yes 👍. Since its both getting the list and updating, it would be nice to have 2 different hooks for that.

Reply:
@Rasim-Deniz

Reply:
I see no reason to pass the candidates down as props. The whole purpose of making the CandidateList container is to isolate all candidate display actions inside it